euphonic/development/02_releases.md

```diff
@@ -3,52 +3,33 @@
-will trigger a new PR for the Conda release in
-https://github.com/conda-forge/euphonic-feedstock.
-
-Currently releases are done from `master` rather than having a release branch.
-This may change in the future as the project evolves.
-
-# Release process
-## 1. Ensure tests pass on all Python versions
-By default Euphonic tests only run on the lowest and highest supported Python
-versions (e.g. 3.7 and 3.10) for each commit. Before a release, all
-intermediate versions should also be tested (e.g. 3.8, 3.9). To run tests on
-the intermediate versions, a
-[workflow dispatch](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch)
-can be used to manually run the tests on the master branch, this will run for
-all supported Python verisons. Make sure these all pass before moving on to
-the next step.
-
-## 2. Update the changelog
-* Update the `Unreleased` title in `CHANGELOG.rst` and the associated Github
-compare to the new version e.g.
-```
-`Unreleased <https://github.com/pace-neutrons/Euphonic/compare/v0.2.2...HEAD>`_
-----------
-```
-becomes:
-```
-`v0.2.3 <https://github.com/pace-neutrons/Euphonic/compare/v0.2.2...v0.2.3>`_
-------
-```
-
-* Add a new `Unreleased` line e.g.
-```
-`Unreleased <https://github.com/pace-neutrons/Euphonic/compare/v0.2.3...HEAD>`_
-----------
-```
-Note the compare in `Unreleased` now compares with the latest version v0.2.3
-not v0.2.2
-
-## 3. Update CITATION.cff
-
-The 'version' and 'date-released' fields should be updated to the new version
-and current date
-
-## 4. Update any new deprecated directives
+The release process for Euphonic has been changed at the beginning of
+2025 to be more streamlined and facilitate pre-releases or bugfixes on
+historical releases. Most steps are performed by Github Actions
+workflows; some of these use scripts located in the *build_utils/*.
+
+The only thing that really cannot be fixed are the packages uploaded
+to PyPI. If we upload a bad release to PyPI, the only solution is to
+mark it as "yanked", increment the version number and release a new
+version. The release workflow includes a lot of tests and
+sanity-checks to reduce the likelihood of this scenario.
+
+# Release process:
+## 0. Consider tidying up metadata (CHANGELOG.rst, CITATION.cff)
+
+If good practice has been followed, CHANGELOG.rst should already have
+a presentable-looking "Unreleased" section; this will automatically be
+used as notes for the new version.
+
+> **NOTE:** pre-releases do not get a CHANGELOG update, as the new
+> features are still officially "Unreleased".
+
+Similarly, CITATION.cff _should_ have been updated with new
+contributors when a significant contribution was made to the project.
+
+There is no need to change the version number yet.
+
+## 1. Update any new deprecated directives
 
 Euphonic uses Numpy docstrings, which recommends use of Sphinx deprecated directives
```

Comment:
Do we want to change the standard here to always deprecate as literally `vx.y.z` and add a sed/grep call to ensure there are none remaining before a release can be pushed.

Reply:
That would make a lot of sense to include this in the version-bumping mechanics.
With a smart enough grep we could check if the version is greater than current release, and set to current release if so.

I also wonder how it interacts with the new deprecated directive from Python 3.13 and backported through typing_extensions. Does sphinx show those in a similar format?

Reply:
Either way this entails another half-day fiddling with Github Actions, so I propose moving to a new Euphonic Issue.